rasa_nlu_examples/classifiers/sparse_sklearn_intent_classifier.py

```diff
@@ -0,0 +1,267 @@
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Train the intent classifier on a data set."""
+
+        num_threads = kwargs.get("num_threads", 1)
+
+        X, y = self.prepare_data(training_data)
+        self.clf = self._create_classifier(num_threads, y)
+
+        with warnings.catch_warnings():
+            # sklearn raises lots of
+            # "UndefinedMetricWarning: F - score is ill - defined"
+            # if there are few intent examples, this is needed to prevent it
+            warnings.simplefilter("ignore")
+            self.clf.fit(X, y)
+
+    def prepare_data(
+        self, training_data: TrainingData
+    ) -> Tuple[scipy.sparse.spmatrix, np.ndarray]:
+        """
+        Converts a rasa TrainingData object into a tuple of a sparse feature
+        matrix and a dense vector of labels.
+        """
+
+        labels = [e.get("intent") for e in training_data.intent_examples]
+
+        if len(set(labels)) < 2:
+            rasa.shared.utils.io.raise_warning(
+                "Can not train an intent classifier as there are not "
+                "enough intents. Need at least 2 different intents. "
+                "Skipping training of intent classifier.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        y = self.transform_labels_str2num(labels)
+        X = scipy.sparse.vstack(
+            [
+                self._get_sentence_features(example)
+                for example in training_data.intent_examples
+            ]
+        )
+
+        return X, y
+
+    @staticmethod
+    def _get_sentence_features(message: Message) -> scipy.sparse.spmatrix:
+        _, sentence_features = message.get_sparse_features(TEXT)
+        if sentence_features is not None:
+            return sentence_features.features
+
+        raise ValueError(
+            "No sentence features present. Not able to train sklearn policy."
+        )
+
+    def _num_cv_splits(self, y: np.ndarray) -> int:
+        folds = self.component_config["max_cross_validation_folds"]
+        return max(2, min(folds, np.min(np.bincount(y)) // 5))
+
+    def _create_classifier(
+        self, num_threads: int, y: np.ndarray
+    ) -> "sklearn.model_selection.GridSearchCV":
+        from sklearn.model_selection import GridSearchCV
+        from sklearn.naive_bayes import BernoulliNB
+
+        alpha = self.component_config["alpha"]
+        binarize = self.component_config["binarize"]
+        fit_prior = self.component_config["fit_prior"]
+        # dirty str fix because sklearn is expecting
+        # str not instance of basestr...
+        tuned_parameters = [
+            {"alpha": alpha, "binarize": binarize, "fit_prior": fit_prior}
+        ]
+
+        # aim for 5 examples in each fold
+
+        cv_splits = self._num_cv_splits(y)
+
+        return GridSearchCV(
```

Comment:
Ah. Now I understand the list in the `config.yml` file. Do we really want to use a GridSearch here? Rasa has GridSearch functionality via the command line and we also have a [hyperopt](https://github.com/RasaHQ/nlu-hyperopt) library. 
With that in mind it feels like overkill to *also* do GridSearch inside of a model.

Reply:
Yeah that makes sense to me to remove grid search. The grid search was what was done in the SklearnIntentClassifier from the the rasa OS code base and I was just trying to make minimal changes to that. Will remove it.